Pull request - Minor edits

hhelgason opened the pull request:
This PR contains minor edits to scripts and config templates. 

- Minor edits to README.md. 
- Outdated .ecf files removed (percent_plot.ecf)
- Ecflow files added (client.py, percentiles.ecf)
- _analysis.py scripts changed to include category of percentile (for a non-linear colorbar on the web)
- Configuration template files updated
- File_transfer.py script updated. On the first day of a given month, the destination file's name now includes "MONTH1". Also, a text file that includes the date the percentiles correspond to is created and transferred. This insures that the date displayed on the web is the correct date.

Commit:
Committing minor edits to scripts and config templates, outdated files removed, ecflow files added:

- Readme updated
- Outdated .ecf files removed (percent_plot.ecf)
- Ecflow files added (client.py, percentiles.ecf)
- _analysis.py scripts changed to include category of percentile (for a non-linear colorbar on the web)
- Configuration template files updated

Commit:
minor changes to file_transfer.py and swe_cdf_creator.py

lizaclark commented on tools/bin/ecflow/processes/main/post_processing/file_transfer.py:
```diff
@@ -23,17 +25,31 @@
-                          (var, date)), os.path.join(dest_loc, 'vic-metdata_%spercentile.nc' % (var)))
+    if date[-2:] == '01':
+        sftp.put(os.path.join(source_loc, 'vic-metdata_%spercentile_%s.nc' %
+                              (var, date)), os.path.join(dest_loc, 'vic-metdata_%spercentile_%s%s.nc' % (var, month, str(1))))
```
Keep lines to 78 characters long.

hhelgason replied:
Fixed.

lizaclark commented on tools/bin/ecflow/processes/main/post_processing/file_transfer.py:
```diff
@@ -23,17 +25,31 @@
+                              (var, date)), os.path.join(dest_loc, 'vic-metdata_%spercentile_%s%s.nc' % (var, month, str(1))))
+    else:
+        sftp.put(os.path.join(source_loc, 'vic-metdata_%spercentile_%s.nc' %
+                              (var, date)), os.path.join(dest_loc, 'vic-metdata_%spercentilei_CURRENT.nc' % (var)))
```
Keep lines to 78 characters long.

hhelgason replied:
Fixed.

lizaclark commented on tools/bin/ecflow/processes/main/post_processing/percentiles.ecf:
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+%include <qsub.h>
```
Isn't this `qsub_build.h` because we're transferring files? Or do we only update that in the operational system?

hhelgason replied:
I figured we'd only update this in the operational system, since this is a workaround that is specific to Hyak.

lizaclark commented on tools/bin/ecflow/processes/main/models/create_global_file.py:
```diff
@@ -37,7 +37,7 @@
 # generate the path to the initial state file
 in_state = os.path.join(state_path, 'state.%s%s%s_00000.nc' %
-                        (start.year, start.month, start.day))
+                        (vic_start_date[:4], vic_start_date[5:7], vic_start_date[8:10]))
```
stop at 78 characters

hhelgason replied:
Fixed.

lizaclark commented on tools/bin/ecflow/processes/main/post_processing/tm_analysis.py:
```diff
@@ -102,35 +102,63 @@
 
 # Dictionary to DataFrame to Dataset
-df = pd.DataFrame(d, columns=["Latitude", "Longitude", "tmpercentile"])
+df = pd.DataFrame(d, columns=["Latitude", "Longitude", "tmpercentile", "category"])
```
try to keep quotation marks consistent -- I like to use `'` for everything, but if you want to use `"`, use it for everything in the same script.

hhelgason replied:
Good point - fixed!

lizaclark reviewed (changes requested):
There are a few things we should probably change in the future. The percentile-to-category mapping could be done with a dictionary and a for loop, `sm_analysis.py`, `swe_analysis.py`, and `tm_analysis.py` can be merged to make a single file. But since we will likely replace these with a less I/O intensive method, I think we should accept as is. For now, please just address the line comments.

Commit:
Committing changes following review:

- create_global_file.py and file_transfer.py contained lines that were > 78 characters long, this was fixed and scripts were made pep8 compliant
- quotation marks were made consistent in (tm, sm, swe)_analysis scripts.

hhelgason commented:
Comments have been addressed!

lizaclark commented on tools/bin/ecflow/processes/main/post_processing/tm_analysis.py:
```diff
@@ -147,7 +147,7 @@
 
 # Dictionary to DataFrame to Dataset
-df = pd.DataFrame(d, columns=["Latitude", "Longitude", "tmpercentile", "category"])
+df = pd.DataFrame(d, columns=['Latitude', 'Longitude', 'tmpercentile', 'category'])
```
Oh, we're so close. This line is too long.

hhelgason replied:
Fixed :)

Commit:
Committing changes following review:

-(tm, sm, swe)_analysis scripts made pep8 compliant.
